near/src/runtime.rs

```diff
@@ -573,17 +579,18 @@ impl RuntimeAdapter for NightshadeRuntime {
         }
     }
 
-    fn num_shards(&self) -> ShardId {
+    fn num_shards(&self) -> NumShards {
         // TODO: should be dynamic.
-        self.genesis_config.block_producers_per_shard.len() as ShardId
+        self.genesis_config.num_block_producer_seats_per_shard.len() as NumShards
     }
 
     fn num_total_parts(&self, parent_hash: &CryptoHash) -> usize {
         let mut epoch_manager = self.epoch_manager.write().expect(POISONED_LOCK_ERR);
         let epoch_id = epoch_manager.get_epoch_id_from_prev_block(parent_hash).unwrap();
-        if let Ok(block_producers) = epoch_manager.get_all_block_producers(&epoch_id, &parent_hash)
+        if let Ok(settlement) =
+            epoch_manager.get_all_block_producers_settlement(&epoch_id, &parent_hash)
```

Comment:
The length of `settlement` is constant throughout live of the system.
We should return that length from the config, not compute an unnecessary constant size array each time and then take its length.
Also that will remove `parent_hash` from the arguments of `num_total_parts`, which is very desirable.

Reply:
++